### ISSUE TITLE
Remove the vendor directory

### ISSUE THREAD

**Issue by gmclelland** [wanze/TemplateEngineTwig#15]
After spending some time trying to fix the TwigExtensions module, I think we should remove the Vendor directory from this module and the TwigExtensions module and rely on Composer only to install all dependencies.

The problem I ran into was that the TwigExtensions module also includes a composer.json that only list "twig/extensions": "~1.5" as a dependency.  The problem with that is that Twig is a dependency of twig/extensions.  So the TwigExtension module was loading a version of Twig in it's module that wasn't the same as the version of Twig in the TemplateEngineTwig module which was causing the PHP errors.  Ideally there should only be one version of Twig loaded.

I know it's a little inconvenient to "composer require twig/twig:1.35" before you install this module, but I think it's for the better.

- This module will be smaller in size because it won't include harikt's code and the Twig library
- You won't have to update this module every time there is a security update in the Twig library (more secure)

Before I typed up this issue, I tried adding harikt's code to the TwigExtensions module:
```js
{
    "name": "justb3a/processwire-twigextensions",
    "type": "pw-module",
    "description": "ProcessWire module that allows customizing twig, e.g. add extensions",
    "keywords": [ "twig-engine", "processwire"],
    "homepage": "https://github.com/justb3a/processwire-twigextensions",
    "license": "GPL-2.0",
    "authors": [
        {
            "name": "ProcessWire Twig Extensions Contributors",
            "homepage": "https://github.com/justb3a/processwire-twigextensions/contributors"
        }
    ],
    "require": {
        "php": ">=5.3.0",
        "hari/pw-module": "~1.0",
        "wanze/template-engine-twig": "~1.0",
        "twig/twig": "~1.35",
        "twig/extensions": "~1.5"
    }
}
```
But I noticed when I ran "composer update" from the TwigExtensions module directory, it would include a site/modules/TemplateEngineFactory and site/modules/TemplateEngineTwig folders as well as the vendor libraries in the Vendor directory.

That means every time there is change in TemplateEngineTwig's Twig library, the TwigExtensions module would also have to keep the same version of Twig in sync.  The module author would also have to delete the newly created "site" folder as well before committing the library changes to git.

This could be prevented if both modules would remove their Vendor director and rely on Composer to have the user install the dependencies in the site root's Vendor directory.

Here's Ryan's thoughts on how to use composer:
- https://processwire.com/blog/posts/composer-google-calendars-and-processwire/
- https://github.com/ryancramerdesign/GoogleClientAPI (no vendor folder here)

Here's Hari's
- https://processwire.com/talk/topic/4958-composer-support-for-processwire/

What do you think?

**Comment by wanze** [comment]
@gmclelland 

I agree that the vendor folder containing twig should be removed from this repository. But I fear doing this with a minor release, as it would break a lot of installs still using the "traditional" way of installing ProcessWire modules. I am planning to rewrite the TemplateEngineFactory and releasing a new major `2` version. Twig will be the only implemented engine I support for the new version. This version will only support installs via composer, which solves all the problems mentioned by you.

> But I noticed when I ran "composer update" from the TwigExtensions module directory, it would include a site/modules/TemplateEngineFactory and site/modules/TemplateEngineTwig folders as well as the vendor libraries in the Vendor directory.

With Hari's `pw-module` you do not have to execute any composer commands in the module directories. If you want to install this module, you can simply run `composer require wanze/template-engine-twig` in your root directory. This will install

* Twig inside the root `vendor` folder
* The TemplateEngineFactory module, automatically copied to `site/modules`
* The TemplateEngineTwig module, automatically copied to `site/modules`

Twig is autoloaded by ProcessWire because it includes the autoloader from composer.

Composer also solves the problem if two modules use the same 3th-party dependency. By using correct version constraints of the dependencies in `composer.json`, composer would refuse to install the module if the required versions are not compatible.

TL;DR Twig will be removed as dependency with TemplateEngineFactory `2.x`. I just have to figure out how I can support both major versions within ProcessWire ;-)
